oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,499 @@
+      cd tutor
+      pip install -e .
+
+2. Follow Tutor's instructions to prepare the default development environment, setting ``TUTOR_ROOT`` so configuration
+   is stored independently, and making sure to answer "no" when asked if this is a production environment:
+
+   .. code-block::
+
+      export TUTOR_ROOT=${WORKSPACE}/tutor_root
+      tutor local quickstart
+      tutor local stop
+      tutor images build --no-cache openedx-dev
+
+3. Because the master branch of ``edx-platform`` is not supported at the time of writing, one must clone Open edX at the
+   latest supported release *tag*:
+
+   .. code-block::
+
+      cd $WORKSPACE
+      git clone -b open-release/koa.1 git@github.com:edx/edx-platform.git
+
+   .. note::
+
+      Note that branches based off of ``master``, or even ``koa.master``, will not work in production.  As noted
+      `in the documentation <https://docs.tutor.overhang.io/configuration.html#running-a-fork-of-edx-platform>`__, "Tutor
+      will attempt to apply security and bug fix patches that might already be included", and possibly fail.
+
+4. Create a docker-compose override file for the development environment:
+
+   .. code-block::
+
+      vim ${TUTOR_ROOT}/env/dev/docker-compose.override.yml
+
+   And configure it so that your local checkout of ``edx-platform`` is bind-mounted in the correct directory by all
+   relevant containers:
+
+   .. code-block::
+
+      version: "3.7"
+      services:
+        lms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        lms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+
+5. Prepare the ``edx-platform`` repository inside the LMS container:
+
+   .. code-block::
+
+      tutor dev run lms bash
+      pip install -r requirements/edx/development.txt
+      npm install
+      openedx-assets build --env=dev
+      exit
+
+6. You'll need to init the dev environment.  (This is taken care of by ``tutor local quickstart`` for the local
+   production environment, and even though the same databases are used for the development environment, apparently the
+   ``GRANT`` command needs to be run again when MySQL runs on a different container.)
+
+   .. code-block::
+
+      tutor dev init
+
+7. And finally, to start the LMS, CMS, and workers in the background:
+
+   .. code-block::
+
+      tutor dev start -d
+
+Code and asset reloading
+````````````````````````
+
+When ``edx-platform`` is mounted as described above, changes made to python files are reloaded immediately.
+
+Template and asset changes are also available immediately, but a browser refresh and/or a Redis cache flush may be
+necessary:
+
+.. code-block::
+
+   tutor dev exec redis bash
+   redis-cli -h redis FLUSHALL
```

Comment:
Are you sure about that? Since Koa, static assets are not cached in Redis but in process cache: https://github.com/overhangio/tutor/blob/6fa1db1e4f4bc8d5933e047fd729361b4ae899fa/tutor/templates/apps/openedx/config/lms.env.json#L71
Thus in production you would only need to restart the uwsgi process to take into account static asset changes. In development, these changes should be taken into account automatically.

Reply:
@regisb, I did not spend too much time digging, but you're right, the Redis flushing may have been a red herring.

I tested something very simple: adding a string to `lms/templates/footer.html`, seeing it updated, then removing it, and seeing it removed.  The first step was automatic.  The second, wasn't.  My first reaction was to flush the cache, and it "worked" - though I was probably hitting the process cache timeout.  I'll remove the comments on cache flushing.  Thanks!